modules/auxiliary/gather/vmware_vcenter_vmdir_ldap.rb

```diff
@@ -84,7 +85,9 @@ def run
       end
 
       print_status("Dumping LDAP data from vmdir service at #{peer}")
-      entries = ldap.search(base: base_dn)
+
+      # A "-" meta-attribute will dump userPassword (hat tip Hynek)
+      entries = ldap.search(base: base_dn, attributes: %w[* + -])
```

Comment:
So did this actually work to dump all known attributes? It was definitely a leap of faith.

Reply:
Semi-relatedly, I started reversing `vmdird` last night and noticed there's quite a lot (more?) going on. 😅

Reply:
seems it did. I tested with ldapsearch and it seemed to be complete.

Reply:
how did you reverse? There is actually a source code  available at github.

Reply:
I started digging into it with Ghidra. But I see there is https://github.com/vmware/lightwave.

Reply:
https://github.com/vmware/lightwave/blob/3bc154f823928fa0cf3605cc04d95a859a15c2a2/vmdir/server/ldap-head/result.c#L647-L654

Nice.

Reply:
Ah, this was the same source repo that [Guardicore](https://www.guardicore.com/2020/04/pwning-vmware-vcenter-cve-2020-3952/) referenced. :)

Reply:
I referenced it in the PR also

Reply:
Yeah, the password hash format. I added a couple other references. Thanks. :)